sklearn/utils/__init__.py

```diff
@@ -590,3 +591,84 @@ def is_scalar_nan(x):
     # Redondant np.floating is needed because numbers can't match np.float32
     # in python 2.
     return bool(isinstance(x, (numbers.Real, np.floating)) and np.isnan(x))
+
+def safe_name_or_class(obj):
+    """Return a representation of the name or class, or raise AttributeError.
+    
+    If the object has a non-None __name__, its safe_str() is returned.
+    Otherwise something similar is done with the object's __class__.
+    Otherwise AttributeError is raised.
+    """
+
+    name = getattr(obj, '__name__', None)
+    if name is not None:
+        return safe_str(name)
+
+    klass = getattr(obj, '__class__', None)
+    if klass is not None:
+        return '{} instance'.format(safe_str(klass))
+    else:
+        raise AttributeError("No __name__ or __class__ attribute.")
+
+def safe_version(transformers, no_transform):
```

Comment:
I would call transformers "callables" and "no_transform" => "no_operation"

Reply:
Also, I would call "safe_version" => "safe_call"

Reply:
`no_transform` is actually the _result_ of the returned function if none of the calls succeeds: it's not a no-op, so I am proposing to call it `default_result` instead.

Reply:
Addressed in https://github.com/lebigot/scikit-learn/commit/6bc2a39fe5e6a9f65c9d54fb442e18a265e82e9a.